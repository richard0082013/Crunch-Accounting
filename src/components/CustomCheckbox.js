import React  from 'react';
import { useField} from 'formik';

const CustomChecbox = ({children, ...props}) => {
  const [field, meta] = useField(props, 'checkbox');
  return (
    <div style={{display: 'flex'}}>
        <label  className='checkbox'>
      {children}
      </label>
    <input className='checkboxInput' type='checkbox' {...field} {...props} />
  
      {meta.touched && meta.error ? (<div className='error'>{meta.error}</div>): null}
    </div>
  )
}
export default CustomChecbox;