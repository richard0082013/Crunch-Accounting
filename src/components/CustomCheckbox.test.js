import React from 'react';
import {shallow} from 'enzyme';
import CustomCheckbox from './CustomCheckbox'
import {Formik} from 'formik';

describe('test checkboxt ',()=>{
  const props = {
    children: "Email Opt Out",
    name: "emailOptOut"
  };
  test('renders without crashing', () => {
    const wrapper = shallow(<Formik><CustomCheckbox {...props}/></Formik>).dive();
    expect(wrapper).toMatchSnapshot();
  });
  test('renders one checkbox with children', () => {
    const wrapper = shallow(<Formik><CustomCheckbox {...props}/></Formik>).dive();
    expect(wrapper.props().children).toBe(props.children);
  })
})

