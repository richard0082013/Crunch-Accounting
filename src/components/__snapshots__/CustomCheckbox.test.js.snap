// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test checkboxt  renders without crashing 1`] = `ShallowWrapper {}`;
